dask_planner/src/parser.rs

```diff
@@ -18,6 +22,170 @@ macro_rules! parser_err {
+                        .map(|e| PySqlArg::new(Some(e.clone()), None))
+                        .collect()),
+                    _ => Ok(vec![]),
+                },
+                None => Err(py_type_err(
+                    "PySqlArg must contain either a standard or custom AST expression",
+                )),
+            },
+        }
+    }
+
+    #[pyo3(name = "getKwargs")]
+    pub fn get_kwargs(&self) -> PyResult<Vec<(String, PySqlArg)>> {
+        match &self.custom {
+            Some(CustomExpr::Nested(kwargs)) => Ok(kwargs.clone()),
+            _ => Ok(vec![]),
+        }
+    }
+
+    #[pyo3(name = "getSqlType")]
+    pub fn get_sql_type(&self) -> PyResult<SqlTypeName> {
+        match &self.custom {
+            Some(custom_expr) => match custom_expr {
+                CustomExpr::Map(_) => Ok(SqlTypeName::MAP),
+                CustomExpr::Multiset(_) => Ok(SqlTypeName::MULTISET),
+                CustomExpr::Nested(_) => Err(py_type_err("Expected Map or Multiset, got Nested")),
+            },
+            None => match &self.expr {
+                Some(Expr::Array(_)) => Ok(SqlTypeName::ARRAY),
+                Some(Expr::Identifier(Ident { .. })) => Ok(SqlTypeName::VARCHAR),
+                Some(Expr::Value(scalar)) => match scalar {
+                    Value::Boolean(_) => Ok(SqlTypeName::BOOLEAN),
+                    Value::Number(_, false) => Ok(SqlTypeName::BIGINT),
+                    Value::SingleQuotedString(_) => Ok(SqlTypeName::VARCHAR),
+                    unexpected => Err(py_type_err(format!(
+                        "Expected string, got {:?}",
+                        unexpected
+                    ))),
+                },
+                Some(Expr::UnaryOp {
+                    op: UnaryOperator::Minus,
+                    expr,
+                }) => match &**expr {
+                    Expr::Value(Value::Number(_, false)) => Ok(SqlTypeName::BIGINT),
+                    unexpected => Err(py_type_err(format!(
+                        "Expected string, got {:?}",
+                        unexpected
+                    ))),
+                },
+                Some(unexpected) => Err(py_type_err(format!(
+                    "Expected array or scalar, got {:?}",
+                    unexpected
+                ))),
+                None => Err(py_type_err(
+                    "PySqlArg must contain either a standard or custom AST expression",
+                )),
+            },
+        }
+    }
+
+    #[pyo3(name = "getSqlValue")]
+    pub fn get_sql_value(&self) -> PyResult<String> {
+        match &self.custom {
+            None => match &self.expr {
+                Some(Expr::Identifier(Ident { value, .. })) => Ok(value.to_string()),
+                Some(Expr::Value(scalar)) => match scalar {
+                    Value::Boolean(value) => Ok(if *value {
+                        "1".to_string()
+                    } else {
+                        "".to_string()
+                    }),
```

Comment:
This is somewhat unclear mapping of booleans to string motivated by the fact that in Python, only empty strings are falsey (e.g. `bool("False") == True`); not sure if this serves as motivation to break up `getSqlValue` into several functions for the various different literal types.